libb64-1.2/bufsiz.patch

```diff
@@ -0,0 +1,26 @@
+diff --git a/include/b64/decode.h b/include/b64/decode.h
+index 12b16ea..6d79ac5 100644
+--- a/include/b64/decode.h
++++ b/include/b64/decode.h
+@@ -22,7 +22,7 @@ namespace base64
+ 		base64_decodestate _state;
+ 		int _buffersize;
+ 
+-		decoder(int buffersize_in = BUFFERSIZE)
++		decoder(int buffersize_in = BUFSIZ)
```

Comment:
Note that this is a functional change. Upstream passes `-DBUFFERSIZE=16777216` by default when building the library:

https://github.com/libb64/libb64/blob/ce864b17ea0e24a91e77c7dd3eb2d1ac4175b3f0/base64/Makefile#L16

Where glibc sets this to 8192:

https://github.com/bminor/glibc/blob/9b91484bee8f6f1bf1e2d26a8df461b553784528/libio/stdio.h#L100

It's not going to break anything - this is intended to be configurable. But what do you think of this?

https://github.com/libb64/libb64/pull/14

If you're good with that, I can try swapping this patch out with a cherry-pick of that.

Reply:
Thanks for the review -- yes please, way better.  You can go ahead and make that fix / change to libb64-1.2 separately, and land that, while I'm navigating landing the header-check script in tw, and then I'll rework this pipeline and branch accordingly....

Reply:
Went on a ride and, with a cleared head, realized this is probably not the most-wolfi approach. I think what upstream had in mind is that `-DBUFFERSIZE` should be explicitly set by any app that uses it. Instead of bending upstream to have a default, I think the better solution here is to let the pipeline pass arbitrary flags. How about we use the generic `opts:` convention that `autoconf/configure` provides?

Reply:
Okay, so you're suggesting that in the test: section, add a cflags variable, or some such, that the header-check pipeline would use, if present?  If so, I think that's reasonable, I guess...

Reply:
To be clear, I don't want to block the addition of this test on this interface, but perhaps we hold off on adding the test to this package until we decide how to do this, since it will set a pattern.

The generic `opts:` input for `autoconf/configure` seems to just let you append arbitrary strings to the pipeline. That would work - we could have:

```yaml
  pipeline:
    - uses: test/headers
       with:
          opts: -DBUFFERSIZE=123456
```

I like the thought of a specific `CFLAGS` better and, well, we already have that interface!

```yaml
test:
  environment:
    environment:
      CFLAGS: -DBUFFERSIZE=123456
  pipeline:
    - uses: test/tw/header-check
```

Reply:
Have a look now, please @dannf ...  Note there there's a separate PR for tw that will have to land, in order to catch and use the new configure-opts option.